packages/pages/src/util/runtime.ts

```diff
@@ -9,21 +11,21 @@ class Runtime {
   version: string;
 
   constructor() {
-    if (typeof process !== "undefined") {
+    if (isDeno) {
+      this.name = "deno";
+      this.version = "";
+      if (isBrowser) {
+        this.version = (window as any).Deno?.version.deno || "";
+      }
+      this.isServerSide = true;
+    } else if (isNode) {
       this.name = "node";
       this.version = process.versions.node;
       this.isServerSide = true;
-    } else if (typeof window !== "undefined" && !("Deno" in window)) {
+    } else {
```

Comment:
Should we check isBrowser here and add a new else condition that sets things to "unknown" or something like that?

Reply:
I considered that and we could do it, but here was my reasoning. Javascript was created for the browser and spent its first 15 years only being run in the browser. Because of that origin it can be assumed that Javascript is being run in the browser, if no node or deno artifacts are detected.